src/Serialization.js

```diff
@@ -152,18 +159,17 @@ exports.setTxBodyMint = setter("mint");
 exports.newMint = () => lib.Mint.new();
 
 exports._bigIntToInt = maybe => bigInt => {
-  try {
-    const str = bigInt.to_str();
-    if (str[0] == "-") {
-      return maybe.just(
-        lib.Int.new_negative(lib.BigNum.from_str(str.slice(1)))
-      );
+  // this is needed because try/catch overuse breaks runtime badly
+  // https://github.com/Plutonomicon/cardano-transaction-lib/issues/875
+  const str = bigInt.to_str();
+  if (/^-[0-9]+$/.test(str)) {
```

Comment:
Same - no overflow checks

Reply:
removed regex as well, as str is constructed from bigInt it is not necessary